hdbt.theme

```diff
@@ -251,6 +251,55 @@ function hdbt_preprocess_paragraph(array &$variables) {
 
     $variables['gallery_slide_thumbnails'] = $slide_thumbnails;
   }
+
+  // Unit search paragraph.
+  if ($paragraph_type == 'unit_search') {
+    $unit_ids = [];
+
+    // Get all unit ids what content producer has selected for the unit search
+    // view and set them as variables for the template.
+    if ($paragraph->hasField('field_unit_search_units')) {
+      $units = $paragraph->field_unit_search_units->getValue();
+      $ids = [];
+      foreach ($units as $unit) {
+        $ids[] = $unit['target_id'];
+        $unit_ids[] = 'tpr_unit:' . $unit['target_id'];
+      }
+      $variables['unit_search_arguments'] = implode(',', $ids);
+    }
+
+    $cached = \Drupal::cache()->get('unit_search' . 'block' . $variables['current_langcode'] . $variables['unit_search_arguments']);
```

Comment:
`$variables['unit_search_arguments']` should have some "initial" value or this whole thing should be wrapped inside the `hasField()` check since the variable might not be defined if the `field_unit_search_units` field doesn't exist.

Reply:
Yeah, moved everything under `hasField(field_unit_search_units)`.